apis/iam/v1alpha1/serviceaccountkey_types.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// ServiceAccountKeyParameters defines parameters for a desired IAM ServiceAccountKey
+// https://cloud.google.com/iam/docs/reference/rest/v1/projects.serviceAccounts.keys
+//
+type ServiceAccountKeyParameters struct {
+	// KeyAlgorithm is an optional user-specified string that specifies the type of key and algorithm
+	// to use for the key. The default is currently a 2048-bit RSA key. However this may change in the future.
+	// Possible values:
+	//   "KEY_ALG_UNSPECIFIED" - Not specified.
+	//   "KEY_ALG_RSA_1024" - 1024-bit RSA key
+	//   "KEY_ALG_RSA_2048" - 2048-bit RSA key
+	// +optional
+	// +immutable
```

Comment:
Hi @turkenh,
We had a discussion with @muvaf on this, and we had decided not to constrain what we allow in the CRD because this parameter is transparent for us, i.e., we directly pass this param to the Cloud API. The idea was Google Cloud API could add new key types & sizes.

Reply:
sgtm!